demisto_sdk/commands/generate_modeling_rules/generate_modeling_rules.py

```diff
@@ -0,0 +1,439 @@
+    FileType,
+)
+from demisto_sdk.commands.common.logger import setup_rich_logging
+from demisto_sdk.commands.common.tools import get_max_version
+
+app = typer.Typer()
+logger = logging.getLogger("demisto-sdk")
+
+
+@app.command(no_args_is_help=True)
+def generate_modeling_rules(
+    mapping: Path = typer.Argument(
+        ...,
+        exists=True,
+        dir_okay=True,
+        resolve_path=True,
+        show_default=False,
+        help=(
+            "The Path to a csv or tsv file containing the mapping for the modeling rules."
+        ),
+    ),
+    raw_event_path: Path = typer.Argument(
+        ...,
+        exists=True,
+        dir_okay=True,
+        resolve_path=True,
+        show_default=False,
+        help=("The path to a raw event from the api call in a json format."),
+    ),
+    output_path: Path = typer.Argument(
+        ...,
+        exists=True,
+        dir_okay=True,
+        resolve_path=True,
+        show_default=False,
+        help=(
+            "A path to the pack you want to generate the modeling rules in. Best practice to put the working pack path"
+        ),
+    ),
+    vendor: str = typer.Argument(
+        default="test",
+        show_default=False,
+        help=("The vendor name of the product"),
+    ),
+    product: str = typer.Argument(
+        default="test",
+        show_default=False,
+        help=("The vendor name of the product"),
+    ),
+    verbosity: int = typer.Option(
+        0,
+        "-v",
+        "--verbose",
+        clamp=True,
+        max=3,
+        show_default=True,
+        help="Verbosity level -v / -vv / .. / -vvv",
+        rich_help_panel="Logging Configuration",
+    ),
+    quiet: bool = typer.Option(
+        False,
+        help="Quiet output - sets verbosity to default.",
+        rich_help_panel="Logging Configuration",
+    ),
+    log_path: Path = typer.Option(
+        None,
+        "-lp",
+        "--log-path",
+        resolve_path=True,
+        show_default=False,
+        help="Path of directory in which you would like to store all levels of logs. If not given, then the "
+        '"log_file_name" command line option will be disregarded, and the log output will be to stdout.',
+        rich_help_panel="Logging Configuration",
+    ),
+    log_file_name: str = typer.Option(
+        "generate_modeling_rules.log",
+        "-ln",
+        "--log-name",
+        resolve_path=True,
+        help="The file name (including extension) where log output should be saved to.",
+        rich_help_panel="Logging Configuration",
+    ),
+):
+    errors = False
+    try:
+        setup_rich_logging(verbosity, quiet, log_path, log_file_name)
+
+        outputfile_schema = output_path.joinpath(
+            f"{vendor}_{product}_modeling_rules.json"
+        )
+        outputfile_xif = output_path.joinpath(f"{vendor}_{product}_modeling_rules.xif")
+        outputfile_yml = output_path.joinpath(f"{vendor}_{product}_modeling_rules.yml")
```

Comment:
instead of joinpath, use `Path(part_1, part_2, Part_3, ... )` or `some_path_object / 'sub_folder_or_file'`